### PR TITLE
Add support for multiple connections in the database and tests

### PR DESCRIPTION
**Pull request by emanueldima**

### CODE REVIEW

File: gefserver/pier/pier.go [hunk 1]
```diff
@@ -380,10 +437,13 @@ func (p *Pier) runJob(job *db.Job, service db.Service, inputPID string, limits d
 	p.updateJobDurationTime(*job)
 }
 
-// RemoveVolumeInUse removes a volume that may seem to be in use
-func (p *Pier) WaitAndRemoveVolume(id dckr.VolumeID) error {
+func (p *Pier) waitAndRemoveVolume(connectionID db.ConnectionID, id dckr.VolumeID) error {
```
Comment (ubuntolog) [inline gefserver/pier/pier.go]: Comment to the function is missing

File: gefserver/server/api.go [hunk 2]
```diff
@@ -166,6 +167,22 @@ func (s *Server) newBuildImageHandler(w http.ResponseWriter, r *http.Request) {
 	Response{w}.Location(loc).Created(jmap("Location", loc, "buildID", buildID))
 }
 
+func (s *Server) getConnectionIDParam(r *http.Request) (db.ConnectionID, error) {
```
Comment (ubuntolog) [inline gefserver/server/api.go]: Comment to the function

File: gefserver/pier/pier.go (hunk 1 above)
Comment (emanueldima) [reply gefserver/pier/pier.go]: The internally used (unexported) functions don't need comments, go vet does not complain about this.

Comment (ubuntolog) [reply gefserver/pier/pier.go]: ok, I was not sure

**Pull request merged**